Libraries/LibC/syslog.cpp

```diff
@@ -0,0 +1,136 @@
+// Has to be defined before including due to legacy Unices
+#define SYSLOG_NAMES 1
+
+#include <stdio.h>
+#include <string.h>
+#include <syslog.h>
+#include <unistd.h>
+
+// This implementation doesn't talk to a syslog server. Any options related to
+// that are no-ops.
+
+extern "C" {
+
+// For implementation simplicity, we actually only use the re-entrant version
+// of each function, and the version that isn't just redirects with a static
+// struct to share.
+static struct syslog_data global_log_data = {
+    .ident = nullptr,
+    .logopt = 0,
+    .facility = LOG_USER,
+    .maskpri = LOG_UPTO(LOG_DEBUG)
+};
+
+// Used when ident is null, since syslog traditionally prints the program's
+// own name; the process name will always be the same unless we exec.
+static char program_name_buffer[256];
+static bool program_name_set = false;
+
+// Convenience function for initialization and checking what string to use
+// for the program name.
+static const char* get_syslog_ident(struct syslog_data* data)
+{
+    if (!program_name_set && data->ident == nullptr)
+        program_name_set = get_process_name(program_name_buffer, 256) >= 0;
+
+    if (data->ident != nullptr)
+        return data->ident;
+    else if (program_name_set)
+        return program_name_buffer;
+
+    return nullptr; // should be never reached
```

Comment:
Maybe call ASSERT_NOT_REACHED(); before the return @awesomekling

Reply:
Yup, that would be good.